### DISCUSSION TITLE
How to use filename of file as a date in a date calculation?

### DISCUSSION THREAD

**Discussion by EpicPilgrim** [SilentVoid13/Templater#1114] [Help]
For my "Daily Notes" template, I have the following in the top:

`<< [[<% tp.date.now("YYYY-MM-DD", -1) %>]] | [[<% tp.date.now("YYYY-MM-DD", 1) %>]] >>
`

This works fine, unless I click on tomorrow's daily note early, to pre-populate it with some information, then the links are obviously based on today's date and now out of context for that note.

My daily notes are already named in the format "YYYY-MM-DD.md". Is there some way I can take the note title/filename and get the previous and next day into Templater from that, instead of based on today's date?

**Comment by Zachatoo** [comment]
There's two more arguments you can pass to `tp.date.now()`, reference and reference format

```js
[[<% tp.date.now("YYYY-MM-DD", -1, tp.file.title, "YYYY-MM-DD") %>|yesterday]]
 
[[<% tp.date.now("YYYY-MM-DD", 1, tp.file.title, "YYYY-MM-DD") %>|tomorrow]]
```

Relevant links:

https://silentvoid13.github.io/Templater/internal-functions/internal-modules/date-module.html#tpdatenowformat-string--yyyy-mm-dd-offset-numberstring-reference-string-reference_format-string

https://zachyoung.dev/snippets/templater#daily-note-links-to-yesterday-and-tomorrow

**Comment by EpicPilgrim** [comment]
Oh amazing. I should have read the docco, I just assumed ".now" wouldn't mean anything than... well... now. Thank you!
